src/UI/Components/Diagram/actions.ts

```diff
@@ -135,30 +154,76 @@ export function appendInstance(
       graph,
       paper,
       instanceAsTable,
-      ServiceInstanceModel,
+      serviceInstanceModel,
       serviceInstance.candidate_attributes,
       "candidate",
+      instanceToConnectRelation,
     );
   } else {
     handleAttributes(
       graph,
       paper,
       instanceAsTable,
-      ServiceInstanceModel,
+      serviceInstanceModel,
       serviceInstance.active_attributes as InstanceAttributeModel,
       "active",
+      instanceToConnectRelation,
     );
   }
-  const appendedInstances = serviceWithReferences.relatedInstances.map(
-    (relatedInstance) => {
-      return appendInstance(paper, graph, relatedInstance, services, false, {
-        id: serviceWithReferences.instance.data.id,
-        name: serviceWithReferences.instance.data.service_entity,
-      });
-    },
-  );
-  connectEntities(graph, instanceAsTable, appendedInstances);
+  serviceWithReferences.relatedInstances.forEach((relatedInstance) => {
+    const isMainInstanceBoolean = false;
```

Comment:
`isMainInstance` already implies it's a boolean :)

Reply:
Yes, I couldn't find better name and `isMainInstance` is used in the function parameters itself :D Do you have any suggestions?

Reply:
If you end up needing the same name, maybe it's worth extracting the logic 🤔 Or maybe it means there is some overcomplication somewhere. Maybe the appendInstance needs to accept a type of instance, rather than a boolean value defining if it's a main instance or not. You'd then have an Enum InstanceType = "main" | "whichever"